src/target/nrf91.c

```diff
@@ -49,6 +93,14 @@ static bool nrf91_flash_erase(target_flash_s *flash, target_addr_t addr, size_t
 	return nrf91_wait_ready(target, NULL);
 }
 
+static bool nrf91_uicr_flash_erase(target_flash_s *flash, target_addr_t addr, size_t len)
+{
+	/* can't erase UICR without a mass-erase */
+	/* TODO: this will silently fail if you actually need to erase */
```

Comment:
Easiest way to fix this would be to `gdb_outf()` a message like "Skipping UICR erase, operation requires mass erase". If there's a good way to blank check the UICR that can be used to have this routine `return true` with no message, and then `return false` if the UICR is not blank.